src/lib.rs

```diff
@@ -136,7 +137,10 @@ impl<T: Clone> Clone for SendCell<T> {
 impl<T> Drop for SendCell<T> {
     fn drop(&mut self) {
         if thread::current().id() != self.thread_id {
-            panic!("trying to convert to inner value in invalid thread");
+            // Explicitly leak the value. Destructors of inner fields are still run after panic.
+            let value = self.value.take();
+            mem::forget(value);
+            panic!("trying to run destructor in invalid thread");
```

Comment:
We might also want to avoid panicking here if the current thread already is panicking. Double panic always kills the process

There is API to check if the current thread is panicking

Reply:
Nah that's fine, it doesn't really matter in the end and I'll wait until someone actually needs this :)